### DISCUSSION TITLE
How to access AppSettings.json

### DISCUSSION THREAD

**Discussion by asurion-ammar** [CoreWCF/CoreWCF#799] [General]
Hi, I was wondering, how can we access appsettings.json in this case. We aren't using any controllers here so DI may not work. What are the alternatives.

**Comment by Kralizek** [comment]
Did you try taking a dependency on `IOptions<T>` in your service? 🤔

**Reply by asurion-ammar** [reply]
Well, I haven't tried it because I don't know if it will work as there is no SuperClass extension in CoreWCF like we ahve Controllers  (where the dependency is injected) when we are writing a Rest service in .NET Core

**Reply by g7ed6e** [reply]
There is a sample available here https://github.com/CoreWCF/samples/tree/main/Scenarios/Logging-and-Dependency-Injection that show how to pull `ILogger` instance from DI either using constructor dependency injection or operation dependency injection. You should be able to adapt this code replacing `ILogger` with `IOptions<T>` or `IConfiguration`.

**Reply by asurion-ammar** [reply]
Thanks, I will definitely try that. Also do you know any sample where the CoreWCF has been deployed in a docker container?
